rapids-cmake/cython/add_rpath_entries.cmake

```diff
@@ -0,0 +1,90 @@
+# the License.
+# =============================================================================
+
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cython_add_rpath_entries
+-------------------------------
+
+.. versionadded:: v22.12.00
+
+Set the RPATH entries for all targets associated with a provided associated target.
+
+.. code-block:: cmake
+
+rapids_cython_add_rpath_entries(
+    TARGET <associated_target>
+    PATHS <path1> <path2> ...
+    [ROOT_DIRECTORY <root-dir>]
+)
+
+This function will affect all targets created up to the point of this call. It
+will have no effect on targets created afterwards.
+
+``TARGET``
+  The associated target for which we are setting RPATH entries. Any target
+  created using :code:command:`rapids_cython_create_modules` with the argument
+  `ASSOCIATED_TARGET associated_target` will have its RPATH entries updated.
+
+``PATHS``
+  The paths to add to the RPATH. Paths may either be absolute or relative to
+  the ROOT_DIRECTORY. The paths are always converted to be relative to the
+  current directory i.e relative to $ORIGIN in the RPATH.
+
+``ROOT_DIRECTORY``
+  The ROOT_DIRECTORY for the provided paths. Defaults to ${PROJECT_SOURCE_DIR}.
+  Has no effect on absolute paths.
+
+#]=======================================================================]
+function(rapids_cython_add_rpath_entries)
+  list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.cython.add_rpath_entries")
+
+  set(options)
+  set(one_value ROOT_DIRECTORY TARGET)
+  set(multi_value PATHS)
+  cmake_parse_arguments(_RAPIDS_CYTHON "${options}" "${one_value}" "${multi_value}" ${ARGN})
+
+  # By default paths are relative to the current project root.
+  if(NOT _RAPIDS_CYTHON_ROOT_DIRECTORY)
+    set(_RAPIDS_CYTHON_ROOT_DIRECTORY "${PROJECT_SOURCE_DIR}")
+  endif()
+
+  # Transform all paths to paths relative to the current directory.
+  set(cleaned_paths)
+  foreach(path IN LISTS _RAPIDS_CYTHON_PATHS)
+    if(NOT IS_ABSOLUTE path)
+      # Convert relative paths to paths from the provided root, then make them relative to the
+      # current directory.
+      cmake_path(ABSOLUTE_PATH path BASE_DIRECTORY "${_RAPIDS_CYTHON_ROOT_DIRECTORY}")
+      cmake_path(ABSOLUTE_PATH path)
+    endif()
+    list(APPEND cleaned_paths "${path}")
+  endforeach()
+
+  get_property(targets GLOBAL PROPERTY "rapids_cython_associations_${_RAPIDS_CYTHON_TARGET}")
+  foreach(target IN LISTS targets)
+    # Compute the path relative to the current target.
+    set(target_paths)
+    get_target_property(target_source_dir ${target} SOURCE_DIR)
+    foreach(target_path IN LISTS cleaned_paths)
+      cmake_path(RELATIVE_PATH target_path BASE_DIRECTORY "${target_source_dir}")
+      list(APPEND target_paths "$ORIGIN/${target_path}")
+    endforeach()
+    list(JOIN target_paths ";" target_paths)
+
+    get_target_property(current_rpath ${target} INSTALL_RPATH)
```

Comment:
You can simplify this by using `set_property(TARGET` with the APPEND mode. That will remove the need to fetch any existing INSTALL_RPATH entries.

Reply:
In general, is there a reason to prefer `set_target_properties` vs `set_property`? I know to always prefer target-specific operations rather than setting global properties, but with the `set_property(TARGET)` syntax it seems like either would be fine. The former supports multiple properties, while the latter supports some other things like `APPEND`, so I guess it varies on a case by case basis. Am I missing anything there?

Reply:
`set_target_properties` has a subset of features compared to `set_property`. Personally I only use `set_property` for `APPEND` as other developers are less framilar with the syntax.